typing/typemode.ml

```diff
@@ -613,6 +570,52 @@ let least_modalities_implying mut (t : Modality.Const.t) =
   in
   exclude_implied @ overridden
 
+let untransl_mod_bounds (bounds : Types.Jkind_mod_bounds.t) : Parsetree.modes =
+  let crossing = Types.Jkind_mod_bounds.crossing bounds in
+  let modality =
+    let open Modality in
+    let add_comonadic acc axis =
+      let (Crossing.Comonadic.Atom.Modality atom) =
+        Crossing.proj (Crossing.Axis.Comonadic axis) crossing
+      in
+      Const.set (Comonadic axis) atom acc
+    in
+    let add_monadic acc axis =
+      let (Crossing.Monadic.Atom.Modality atom) =
+        Crossing.proj (Crossing.Axis.Monadic axis) crossing
+      in
+      Const.set (Monadic axis) atom acc
+    in
+    let add acc = function
+      | Value.Axis.P (Comonadic axis) -> add_comonadic acc axis
+      | Value.Axis.P (Monadic axis) -> add_monadic acc axis
+    in
+    List.fold_left add Modality.Const.id Value.Axis.all
+  in
+  let modality_annots =
+    least_modalities_implying Types.Immutable modality
+    |> List.map (fun atom ->
+           let { Location.txt = Parsetree.Modality s; _ } =
+             untransl_modality atom
+           in
+           { Location.txt = Parsetree.Mode s; loc = Location.none })
+  in
```

Comment:
I think you can make `mode_intf.mli` expose the type definition of `Crossing.Monadic.t` and `Crossing.Comonadic.t` and use them to acquire the modality stored inside a `Crossing.t`, then use that to construct `Modality.Const.t`(again, by exposing the type declaration of it), then just call `untransl_modalities`. That should be shorter code.